src/binary-writer.cc

```diff
@@ -245,7 +245,7 @@ void BinaryWriter::WriteBlockDecl(const BlockDeclaration& decl) {
   Index index = decl.has_func_type ? module_->GetFuncTypeIndex(decl.type_var)
                                    : module_->GetFuncTypeIndex(decl.sig);
   assert(index != kInvalidIndex);
-  WriteS32Leb128(stream_, index, "block type function index");
+  WriteU32Leb128WithReloc(index, "block type function index", RelocType::TypeIndexLEB);
```

Comment:
I think this should be conditional on `decl.has_func_type`?   Normal block with a single return value don't need relocations, right?

Reply:
I haven't checked the spec, but from the existing code I assumed that we need a type with reloc for all blocks that have more than zero params or more than one result. The method returns early in these cases.

Reply:
There are two type is block signature ... either the block has a basic value type .. or it has a function type.  You can see this being checked above with `decl.has_func_type`.

If you do this unconditionally then *every* block will end up with a relocation.  This is not what we want.  Ordinary block (all blocks that predate the multi value proposal) should *not* have relocations for their types.

Its not just an optimization is simply won't work.  Each TypeIndexLEB relocation must correspond for an entry in the type section, and there are no such entry for blocks with value types.

Reply:
Apologies I didn't fully read your comment.  You are correct, regular blocks should return early here and this path should only be for multivalue enabled blocks.